### PR TITLE
API Gateway Setup

### PR DESCRIPTION
**Pull request by clintonb**
ECOM-3862

### CODE REVIEW

**Comment by clintonb** [comment]
YAML validated by the fact this all works.
@peter-fogg @bderusha please review the acceptance tests

**Comment by peter-fogg** [comment]
:+1:

File: acceptance_tests/test_api_gateway.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    @ddt.data(*PATHS)
+    def test_endpoint_ok(self, path):
+        """ Verify the endpoint returns HTTP 200 for valid requests. """
+        headers = {
+            'Authorization': 'Bearer {token}'.format(token=DISCOVERY_API_ACCESS_TOKEN)
+        }
+        self.assert_api_response(path, **headers)
+
+    @ddt.data(*PATHS)
+    def test_endpoint_not_authorized(self, path):
```
Comment (bderusha) [inline acceptance_tests/test_api_gateway.py:43]: Should we also test for 500 errors?

**Comment by bderusha** [comment]
:+1: Only one question about how much coverage we need on this (aka which status codes we need to test)

File: acceptance_tests/test_api_gateway.py (hunk 1 above)
Comment (clintonb) [inline acceptance_tests/test_api_gateway.py:43]: Nope. We should not expect to receive HTTP 500 errors. Creating an endpoint that purposely fails solely for testing is a bad thing in the wrong hands.

**Pull request merged**